confAllChambers.py

```diff
@@ -66,16 +72,38 @@ def launchArgs(shelf,slot,link,run,armDAC,armDACBump,config,cName,debug=False):
     parser = argparse.ArgumentParser(description="Tool for configuring all front-end electronics")
     parser.add_argument("--armDAC", type=int,default = 100,help="CFG_THR_ARM_DAC value to write to all VFATs")
     parser.add_argument("--armDACBump", type=int,help="CFG_THR_ARM_DAC value for all VFATs", default=0)
-    parser.add_argument("--config", action="store_true",help="Set Configuration from simple txt files")
     parser.add_argument("-d","--debug", action="store_true",help="prints additional debugging information")
     parser.add_argument("--run", action="store_true",help="Set VFATs to run mode")
     parser.add_argument("--series", action="store_true",help="Run tests in series (default is false)")
     parser.add_argument("--shelf", type=int,help="uTCA shelf number",default=1)
+
+    confGroup = parser.add_argument_group(title="Configuration Group",description="Options for configuring channel registers and CFG_THR_ARM_DAC") 
+    confGroup.add_argument("--applyMasks", action="store_true",help="If paired with --chConfig channel masks defined in chConfig text file will be applied; otherwise no effect")
+    confGroup.add_argument("--vfatConfig", action="store_true",help="Set only CFG_THR_ARM_DAC registers from symlinks found under $DATA_PATH/configs")
+    chConfGroup = confGroup.add_mutually_exclusive_group()
+    chConfGroup.add_argument("--chConfig", action="store_true",help="Set only channel registers from symlinks found under $DATA_PATH/configs")
+    chConfGroup.add_argument("--zeroChan", action="store_true",help="Zeros all channel registers")
     args = parser.parse_args()
 
     from gempython.utils.wrappers import envCheck
     envCheck('DATA_PATH')
 
+    # determine configType, 4-bit number with bit meaning as:
+    # [0] -> apply vfatConfig
+    # [1] -> apply chConfig
+    # [2] -> apply channel masks
+    # [3] -> zero channels
+    configType=0x0
+    if args.vfatConfig:
+        configType += 0x1
+    if args.chConfig:
+        configType += (0x1 << 1)
```

Comment:
Similarly, here you can use:
```python
configType |= 0x10
```

You could define a set of enums that define the meaning of each bit, so that both here and above, the intent is clear from the code (but this is not strictly necessary)

Reply:
Updated using bitwise operators.

> You could define a set of enums that define the meaning of each bit, so that both here and above, the intent is clear from the code (but this is not strictly necessary)

If this is not necessary I would prefer to wait on this for either migration to python3 when the `enum` type will be available and/or when we move to `boost`ing things from `cmsgemos` and have this in a dedicated header file in a single location.